core/src/impl/Kokkos_HostThreadTeam.hpp

```diff
@@ -864,55 +864,87 @@ KOKKOS_INLINE_FUNCTION
 
 //----------------------------------------------------------------------------
 
-template <typename iType, class Closure, class Member>
+template <typename iType, class Closure, class Member, typename ValueType>
 KOKKOS_INLINE_FUNCTION
-    std::enable_if_t<Impl::is_host_thread_team_member<Member>::value>
+    std::enable_if_t<!Kokkos::is_reducer<ValueType>::value &&
+                     Impl::is_host_thread_team_member<Member>::value>
     parallel_scan(Impl::TeamThreadRangeBoundariesStruct<iType, Member> const&
                       loop_boundaries,
-                  Closure const& closure) {
-  // Extract ValueType from the closure
-
-  using value_type = typename Kokkos::Impl::FunctorAnalysis<
+                  Closure const& closure, ValueType& return_val) {
+  // Extract ValueType from the Closure
+  using ClosureValueType = typename Kokkos::Impl::FunctorAnalysis<
       Kokkos::Impl::FunctorPatternInterface::SCAN, void, Closure,
-      void>::value_type;
-
-  value_type accum = 0;
+      ValueType>::value_type;
+  static_assert(std::is_same<ClosureValueType, ValueType>::value,
+                "Non-matching value types of closure and return type");
 
   // Intra-member scan
   for (iType i = loop_boundaries.start; i < loop_boundaries.end;
        i += loop_boundaries.increment) {
-    closure(i, accum, false);
+    closure(i, return_val, false);
```

Comment:
That does not look right.  The Kokkos semantics for reductions and scan with return value is we ignore the "initial" value of the argument that is passed by the user and overwrite with the result.

Reply:
@thearusable here is the link: https://kokkos.github.io/kokkos-core-wiki/API/core/parallel-dispatch/parallel_scan.html#semantics

i think we need to use a local variable `accum` as before, and then assign 
```
value_type accum = 0;
... 
return_value = accum; 
```

Reply:
Ok, Fixed